tests/hil/tests/connection/test.c

```diff
@@ -1,16 +1,55 @@
 #include <golioth/client.h>
 #include <golioth/golioth_sys.h>
+#include <golioth/rpc.h>
 
 LOG_TAG_DEFINE(test_connect);
 
+static golioth_sys_sem_t sleep_temporarily_sem;
+static int sleep_temporarily_msec;
+
+static enum golioth_rpc_status on_stop_temporarily(zcbor_state_t *request_params_array,
+                                                   zcbor_state_t *response_detail_map,
+                                                   void *callback_arg)
+{
+    double seconds;
+    bool ok;
+
+    ok = zcbor_float_decode(request_params_array, &seconds);
+    if (!ok)
+    {
+        GLTH_LOGE(TAG, "Not okay!");
+        return GOLIOTH_RPC_INVALID_ARGUMENT;
+    }
+
+    sleep_temporarily_msec = seconds * 1000.0;
+    golioth_sys_sem_give(sleep_temporarily_sem);
+
+    GLTH_LOGI(TAG, "Received stop request for %d", (int) seconds);
+
+    return GOLIOTH_RPC_OK;
+}
+
 void hil_test_entry(const struct golioth_client_config *config)
 {
     struct golioth_client *client = golioth_client_create(config);
+    struct golioth_rpc *grpc = golioth_rpc_init(client);
+
+    sleep_temporarily_sem = golioth_sys_sem_create(1, 0);
+
+    golioth_rpc_register(grpc, "stop_temporarily", on_stop_temporarily, NULL);
 
     (void) client;
 
     while (1)
     {
-        golioth_sys_msleep(5000);
+        golioth_sys_sem_take(sleep_temporarily_sem, GOLIOTH_SYS_WAIT_FOREVER);
+
+        GLTH_LOGI(TAG, "Stopping client");
+        golioth_client_stop(client);
+
+        golioth_sys_msleep(sleep_temporarily_msec);
```

Comment:
`sleep_temporarily_msec` could be uninitialized here if the RPC hasn't run yet, right?

Reply:
We would not reach this line, since we waited for semaphore that was released in RPC. So `sleep_temporarily_msec` was guaranteed to be set at this point. Anyway, this is outdated without RPC use.